src/lib/comms/sol-network-linux.c

```diff
@@ -236,35 +236,36 @@ _on_event(void *data, int nl_socket, unsigned int cond)
         return false;
     }
 
-    status = recvmsg(nl_socket, &msg, MSG_WAITALL);
-    if (status <= 0) {
-        if (errno == EWOULDBLOCK || errno == EAGAIN)
-            return true;
+    while ((status = recvmsg(nl_socket, &msg, MSG_WAITALL))) {
+        if (status < 0) {
+            if (errno == EWOULDBLOCK || errno == EAGAIN)
```

Comment:
well, EAGAIN basically does it, but first return to mainloop.  I'll put explicitly an continue for this error. Thanks for pointing it.